operators/src/source/gdal_source/mod.rs

```diff
@@ -420,11 +434,48 @@ impl GdalRasterLoader {
         tile_information: TileInformation,
         tile_time: TimeInterval,
     ) -> Result<RasterTile2D<T>> {
-        crate::util::spawn_blocking(move || {
-            Self::load_tile_data(&dataset_params, tile_information, tile_time)
-        })
+        let is_vsi_curl = dataset_params.file_path.starts_with("/vsicurl/");
+
+        retry(
+            dataset_params
+                .retry
+                .map(|r| r.max_retries)
+                .unwrap_or_default(),
+            GDAL_RETRY_INITIAL_BACKOFF_MS,
+            GDAL_RETRY_EXPONENTIAL_BACKOFF_FACTOR,
+            move || {
+                let ds = dataset_params.clone();
+                let file_path = ds.file_path.clone();
+
+                async move {
+                    let load_tile_result = crate::util::spawn_blocking(move || {
+                        Self::load_tile_data(&ds, tile_information, tile_time)
+                    })
+                    .await
+                    .context(crate::error::TokioJoin);
+
+                    match load_tile_result {
+                        Ok(Ok(r)) => Ok(r),
+                        Ok(Err(e)) | Err(e) => {
+                            if is_vsi_curl {
+                                // clear the VSICurl cache, to force GDAL to try to re-download the file
+                                // otherwise it will assume any observed error will happen again
+                                unsafe {
+                                    if let Some(Some(c_string)) =
+                                        file_path.to_str().map(|s| CString::new(s).ok())
+                                    {
+                                        VSICurlPartialClearCache(c_string.as_ptr());
+                                    }
+                                }
```

Comment:
1. This should maybe be part of rust-gdal.
2. Can we test this?

Reply:
1. it could be, but I didn't want to force Johannes to make a release to save half a line in Geo Engine
2. it's tested implicitly in the sentinel2 provider. The fact that it runs through, shows, that gdal retries. I don't really think it makes sense to write a test that ensures that it doesn't work without this line.

Reply:
But shouldn't this offer a good test? You basically inserted this because there are certain errors, where GDAL would return the same result even if the source changed. And this line avoids that, right?

Reply:
b8829790a8bbe5ed8debae4a36e920ab5ec93cea